frontend/www/js/omegaup/components/contest/Report.vue

```diff
@@ -1,67 +1,100 @@
 <template>
-  <div class="panel panel-default">
-    <div class="panel-heading">
+  <div class="card">
+    <div class="card-header">
       {{ ui.formatString(T.contestReport, { contest_alias: contestAlias }) }}
     </div>
-    <div class="panel-body">
-      <div v-for="contestantData in contestReport">
+    <div class="card-body">
+      <div
+        v-for="contestantData in contestReport"
+        :key="contestantData.username"
+        class="pb-2"
+      >
         <h1>{{ T.username }}: {{ contestantData.username }}</h1>
         <h3>
-          {{ T.wordsTotal }}:<span
-            v-if="Object.prototype.hasOwnProperty.call(contestantData, 'total') &amp;&amp; Object.prototype.hasOwnProperty.call(contestantData.total, 'points')"
-            >{{ contestantData.total.points }}</span
-          ><span v-else>0</span>
+          {{ T.wordsTotal }}:
+          <span v-if="hasTotalAndPointsProperties(contestantData)">
+            {{ contestantData.total.points }}
+          </span>
+          <span v-else>0</span>
         </h3>
-        <div v-for="item in contestantData.problems">
+        <div
+          v-for="item in contestantData.problems"
+          :key="`${contestantData.username}_${item.alias}`"
+          class="pb-2"
+        >
           <h3>{{ T.wordsProblem }}: {{ item.alias }}</h3>
           <h3>{{ T.wordsPoints }}: {{ item.points }}</h3>
           <div v-if="item.run_details">
             <template v-for="group in item.run_details.details.groups">
               <table
-                v-if="item.run_details &amp;&amp; (((item || {}).run_details || {} ).details || {} ).groups"
+                v-if="
+                  item.run_details &&
+                  (((item || {}).run_details || {}).details || {}).groups
```

Comment:
```suggestion
                  item?.run_details?.groups
```

qué bueno que ya podemos usar optional chaining!

Reply:
Y mucho más claro. Gracias por la observación. Aunque aquí ya no sería necesario agregar el optional chaining porque ya hay una condición arriba